tableauserverclient/server/endpoint/favorites_endpoint.py

```diff
@@ -0,0 +1,98 @@
+from .endpoint import Endpoint, api
+from .exceptions import MissingRequiredFieldError
+from .. import RequestFactory, WorkbookItem, ViewItem, ProjectItem, DatasourceItem
+from ..pager import Pager
+import xml.etree.ElementTree as ET
+import logging
+import copy
+
+logger = logging.getLogger('tableau.endpoint.favorites')
+
+
+class Favorites(Endpoint):
+    @property
+    def baseurl(self):
+        return "{0}/sites/{1}/favorites".format(self.parent_srv.baseurl, self.parent_srv.site_id)
+
+    # Gets all favorites
+    @api(version="2.5")
+    def get(self, user_item, req_options=None):
```

Comment:
Thanks for the quick turnaround!

Revisiting your question about should there be a favorite_item class: I think so.

This xml parsing logic is fairly harmless here, but if we ever swapped to JSON or something like that, we'd be leaking that into the endpoint, rather than keeping it contained in just the models.

I can think of 3 ways to do this:
1. Have a FavoriteItem class per content type `WorkbookFavoriteItem.from_request` etc
2. Have a single `FavoriteItem` with 4 parsing methods `FavoriteItem.workbooks_from_request`
3. Have a single `FavoriteItem` with 1 parsing method that returns a list of lists, or a dict of lists, or a series of tuples, `FavoriteItem.from_request`

@jorwoods I'm not sure which is best, do you have any thoughts? (You may look to TaskItem for inspiration, I think it has a similar wrapper pattern)

Reply:
Agreed that it likely need to be pulled out of endpoint. The way I'm calling out to the class methods for each of the types feels flimsy too, but I could not see a better path that didn't involve altering a few of the classes substantially.

1.  This feels like it would fracture the favorites endpoint substantially. It also wouldn't reflect the REST API that returns all objects. May be preferable though? May be the clearest option as a python user decides they only want favorites of a certain type. REST API though would return everything, and the object would just filter it down.

2. Again, the REST API returns all objects, so while having individual dispatch is clearer from a python standpoint, there would have to be something calling out to each method.

3. This would probably be my preference, with a dict of lists.  Would sort the objects into easily addressable sets, while also being consistent with the behavior of the REST API.

Reply:
#3 sounds best to me as well.

(You could also have a favorites object with properties for each. This is what Permissions wraps as a list of PermissionRules... but this case is simpler and that'd be no better than a dict.)

So...

```
class FavoritesList(TypedDict):
  workbooks: List[WorkbookItem]
  flows: List[FlowItem]
  datasources: List[DatasourceItem]
  views: List[Views]
```

Reply:
To clarify, because of the snippet you posted: You have `FavoritesList` inherit from TypedDict. Should I put in type hinting? Or was that purely for demonstrative purposes?

Reply:
And this invites other questions: Would it be best for them to be accessed via __getattr__ or __getitem__? A simple enough hack could make it both.

If a user has no favorites of a given item, should it return None, or an empty list? Alternatively it could just not be present on the object, but I think None or empty list is better.

Reply:
Just illustrative, we don't have type hinting in the code base yet.

If it wasn't clear, I think a simple dict is the way to go. No need for the wrapper object.

`None` or `[]` ... I like the idea of an empty list so for loops work just as well. I don't think we're always consistent here throughout the library though.

```
for fav in favorites.workbooks:
    print(fav.id)
    # if empty nothing happens, but no error
```

Reply:
Marking this as resolved as it now has tests and writes to a dictionary as requested.